CMakeLists.txt

```diff
@@ -1,18 +1,23 @@
-if (CMAKE_BUILD_TYPE STREQUAL "Debug")
-    set (SSL_ROOT_PATH ${CMAKE_CURRENT_LIST_DIR}/no-asm)
+
+if(CMAKE_BUILD_TYPE STREQUAL "Debug")
+  set(SSL_ROOT_PATH ${CMAKE_CURRENT_LIST_DIR}/no-asm)
 else()
-    set (SSL_ROOT_PATH ${CMAKE_CURRENT_LIST_DIR})
+  set(SSL_ROOT_PATH ${CMAKE_CURRENT_LIST_DIR})
 endif()
 
-if (Qt6_VERSION VERSION_GREATER_EQUAL 6.5.0)
-    list(APPEND ANDROID_EXTRA_LIBS
-        ${SSL_ROOT_PATH}/ssl_3/${CMAKE_ANDROID_ARCH_ABI}/libcrypto_3.so
-        ${SSL_ROOT_PATH}/ssl_3/${CMAKE_ANDROID_ARCH_ABI}/libssl_3.so)
+if(Qt6_VERSION VERSION_GREATER_EQUAL 6.5.0)
+
+  if(NOT OPENSSL_ROOT_DIR)
+    set(OPENSSL_ROOT_DIR ${SSL_ROOT_PATH}/ssl_3/${CMAKE_ANDROID_ARCH_ABI})
+  endif()
+  list(APPEND ANDROID_EXTRA_LIBS
+       ${SSL_ROOT_PATH}/ssl_3/${CMAKE_ANDROID_ARCH_ABI}/libcrypto_3.so
+       ${SSL_ROOT_PATH}/ssl_3/${CMAKE_ANDROID_ARCH_ABI}/libssl_3.so)
 else()
-    list(APPEND ANDROID_EXTRA_LIBS
-        ${SSL_ROOT_PATH}/ssl_1.1/${CMAKE_ANDROID_ARCH_ABI}/libcrypto_1_1.so
-        ${SSL_ROOT_PATH}/ssl_1.1/${CMAKE_ANDROID_ARCH_ABI}/libssl_1_1.so)
+  if(NOT OPENSSL_ROOT_DIR)
+    set(OPENSSL_ROOT_DIR ${SSL_ROOT_PATH}/ssl_1.1/${CMAKE_ANDROID_ARCH_ABI})
+  endif()
+  list(APPEND ANDROID_EXTRA_LIBS
+       ${SSL_ROOT_PATH}/ssl_1.1/${CMAKE_ANDROID_ARCH_ABI}/libcrypto_1_1.so
+       ${SSL_ROOT_PATH}/ssl_1.1/${CMAKE_ANDROID_ARCH_ABI}/libssl_1_1.so)
 endif()
-
-set_target_properties(${PROJECT_NAME} PROPERTIES
-    QT_ANDROID_EXTRA_LIBS "${ANDROID_EXTRA_LIBS}")
```

Comment:
I'm not a cmake expert, but I think the `QT_ANDROID_EXTRA_LIBS` is still needed.

Reply:
It is not needed in all cases. Also this fails if there is no target named after `${PROJECT_NAME}` which is not always the case. 

Some consider it actually bad practice to reuse the `${PROJECT_NAME}` for target name, even though it is used frequently. 
As there is also the `android_openssl.cmake` file which has a function to assign the libs to the target properties.